### PR TITLE
Enable Azure on .NET Fx

### LINKED ISSUE #2833: ASP.NET intrumentation failed on .NET framework
**Describe the bug**
ASP.NET intrumentation failed sometimes on .NET framework. 
I have 2 webapi ASP.NET applications. One is `WebApiMySql80Netfx462`; the other is `WebApiMySql81Netfx462`. I accessed the first application and saw the ASP.NET instrumentation succeeded. after that, when I tried to access the second application, I did not see any ASP.NET trace/span got collected. Looks like the second ASP.NET instrumentation failed. If I access the second web app in the first place, its traces/spans can be collected.

Attached please find logs:
[logs_aspnet_instrumentation_failure.zip](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/files/12326848/logs_aspnet_instrumentation_failure.zip)

**Runtime environment (please complete the following information):**

- OpenTelemetry Automatic Instrumentation version: 1.0.0-rc.2
- OS: Windows Server 2012 R2
- .NET version: .NET Framework 4.6.2

### PR DESCRIPTION
**Pull request by pjanotti**
## Why

Avoid issues due to incorrect version of S.D.DS on .NET Fx

Fix #2833 and #2796

## What

Enable Azure SDK instrumentation on .NET Framework.

## Tests

Locally verified .NET Framework w/ Linux containers.

## Checklist

- [X] `CHANGELOG.md` is updated.
- [X] Documentation is updated.
- [X] New features are covered by tests.

### CODE REVIEW

File: src/OpenTelemetry.AutoInstrumentation/Configurations/TracerInstrumentation.cs
Comment (Kielek) [inline src/OpenTelemetry.AutoInstrumentation/Configurations/TracerInstrumentation.cs:1]: Please fix also test\OpenTelemetry.AutoInstrumentation.Tests\Configurations\SettingsTests.cs. Around line 254

File: test/IntegrationTests/AzureTests.cs [hunk 1]
```diff
@@ -48,6 +46,10 @@ public void SubmitsTraces(string packageVersion)
         collector.Expect("System.Net.Http");
 #elif NET6_0_OR_GREATER
         collector.Expect("OpenTelemetry.Instrumentation.Http.HttpClient");
+#elif NETFRAMEWORK
+        // On .NET Framework the "OpenTelemetry.Instrumentation.Http.HttpWebRequest"
+        // ends up being suppressed by the addition of headers via the Azure instrumentation
+        // See https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs#L279-L284
```
Comment (Kielek) [inline test/IntegrationTests/AzureTests.cs:52]: Link to main can be desynchronized. Please use commit or tag

```suggestion
        // See https://github.com/open-telemetry/opentelemetry-dotnet/blob/f6a1c04e8115a828dd33269f639daf2924796bae/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs#L279-L284
```

File: docs/manual-instrumentation.md
Comment (Kielek) [inline docs/manual-instrumentation.md:1]: I would keep it as separate issue/PR. I think it is unrealed to Azure NET Fx. fix
What is more, it should be part of #2851. 1.6.0 will force update to 7.0.2.

File: CHANGELOG.md
Comment (Kielek) [inline CHANGELOG.md:1]: Please update also docs\config.md line ~131

File: src/OpenTelemetry.AutoInstrumentation/Configurations/TracerInstrumentation.cs
Comment (pjanotti) [reply src/OpenTelemetry.AutoInstrumentation/Configurations/TracerInstrumentation.cs:1]: 0a5f4bf144f7de7214f979f36a38b89d9f969666

File: docs/manual-instrumentation.md
Comment (pjanotti) [reply docs/manual-instrumentation.md:1]: Yeah better as a separate PR, it deserves its own entry on the troubleshooting.

File: src/OpenTelemetry.AutoInstrumentation/Configurations/TracerInstrumentation.cs
Comment (zacharycmontoya) [reply src/OpenTelemetry.AutoInstrumentation/Configurations/TracerInstrumentation.cs:1]: And also src\OpenTelemetry.AutoInstrumentation\Instrumentation.cs line 305, I believe

**Review by zacharycmontoya — COMMENTED** [review]
With the changes Piotr suggested, I think the PR will be ready to merge

File: test/IntegrationTests/AzureTests.cs (hunk 1 above)
Comment (pjanotti) [reply test/IntegrationTests/AzureTests.cs]: 913501fd894e6e02dfe64d9bc0732979c7dcc1ce

File: docs/manual-instrumentation.md
Comment (pjanotti) [reply docs/manual-instrumentation.md:1]: 2632e95c4c9afce4b10cb113d23e196de61b258b

File: CHANGELOG.md
Comment (pjanotti) [reply CHANGELOG.md:1]: 47b848f79821b842ad70b59bf0a60204ceec2682

**Review by nrcventura — APPROVED** [review]

File: src/OpenTelemetry.AutoInstrumentation/Configurations/TracerInstrumentation.cs
Comment (pjanotti) [reply src/OpenTelemetry.AutoInstrumentation/Configurations/TracerInstrumentation.cs:1]: Missed your comment earlier @zacharycmontoya - looking...

**Review by Kielek — APPROVED** [review]

**Pull request merged**